cmd/storage-datatypes.go

```diff
@@ -151,6 +151,9 @@ type FileInfo struct {
 	MarkDeleted                   bool // mark this version as deleted
 	DeleteMarkerReplicationStatus string
 	VersionPurgeStatus            VersionPurgeStatusType
+
+	// purposefully kept lower case such that 'msgp' ignores this field.
+	data []byte // optionally carries object data
```

Comment:
The way it is done we might just as well serialize it as part of the FileInfo struct. That will simplify the code a bunch.

```
//go:generate msgp -unexported -file=$GOFILE
```

Will add unexported fields as well.